lib/Rex/Commands/Iptables.pm

```diff
@@ -548,4 +610,25 @@ sub _rule_exists {
   return 0;
 }
 
+sub _get_ip_version {
+  my ($params) = @_;
+  if ( defined $params->[0] && !ref $params->[0] ) {
+    if ( $params->[0] eq "-4" || $params->[0] eq "-6" ) {
+      return shift @$params;
+    }
+  }
+  return -4;
+}
+
+sub _get_executable {
+  my ($params) = @_;
+  return _get_ip_version($params) == -6 ? "ip6tables" : "iptables";
```

Comment:
I think we can extend this subroutine to return the full path of the executable, instad of just a name. Like:
```perl
my $binary = _get_ip_version($params) == -6 ? "ip6tables" : "iptables";
return can_run($binary);
```

This way the check in the `iptables` subroutine (line #340) might also be simplified. I'll leave a note there too.